frontend/src/lib/k8s/pod.ts

```diff
@@ -117,6 +134,151 @@ class Pod extends makeKubeObject<KubePod>('Pod') {
+    for (const condition of conditions) {
+      if (condition.type === 'Ready' && condition.Status === 'True') {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // Implementation based on: https://github.com/kubernetes/kubernetes/blob/33bf8a8ebd33eae0d45c18feb1c05baa6a055cde/pkg/printers/internalversion/printers.go#L558
+  getDetailedStatus(): PodDetailedStatus {
+    // We cache this data to avoid going through all this logic when nothing has changed
+    if (
+      !!this.detailedStatusCache.details &&
+      this.detailedStatusCache.resourceVersion === this.jsonData.metadata.resourceVersion
+    ) {
+      return this.detailedStatusCache.details;
+    }
+
+    // We cache this data to avoid going through all this logic when nothing has changed
+    if (
+      !!this.detailedStatusCache.details &&
+      this.detailedStatusCache.resourceVersion === this.jsonData.metadata.resourceVersion
+    ) {
+      return this.detailedStatusCache.details;
+    }
+
+    let restarts = 0;
+    const totalContainers = (this.spec.containers ?? []).length;
+    let readyContainers = 0;
+    let message = '';
+    let lastRestartDate = new Date(0);
+
+    let reason = this.status.reason || this.status.phase;
+
+    let initializing = false;
+    for (const i in this.status.initContainerStatuses) {
+      const container = this.status.initContainerStatuses[i];
+      restarts += container.restartCount;
+      lastRestartDate = this.getLastRestartDate(container, lastRestartDate);
+
+      switch (true) {
+        case container.state.terminated?.exitCode === 0:
+          continue;
+        case !!container.state.terminated:
+          if (!container.state.terminated.reason) {
+            if (container.state.terminated.signal !== 0) {
+              reason = `Init:Signal:${container.state.terminated.signal}`;
+            } else {
+              reason = `Init:ExitCode:${container.state.terminated.exitCode}`;
+            }
+          } else {
+            reason = 'Init:' + container.state.terminated.reason;
+          }
+          message = container.state.terminated.message || '';
+          initializing = true;
+          break;
+        case !!container.state.waiting?.reason &&
+          container.state.waiting.reason !== 'PodInitializing':
+          reason = 'Init:' + container.state.waiting.reason;
+          initializing = true;
+          message = container.state.waiting.message || '';
+          break;
+        default:
+          reason = `Init:${i}/${(this.spec.initContainers || []).length}`;
+          initializing = true;
+      }
+      break;
+    }
+
+    if (!initializing) {
+      restarts = 0;
+      let hasRunning = false;
+      for (let i = (this.status?.containerStatuses?.length || 0) - 1; i >= 0; i--) {
+        const container = this.status?.containerStatuses[i];
+
+        restarts += container.restartCount;
+        lastRestartDate = this.getLastRestartDate(container, lastRestartDate);
```

Comment:
LastRestartDate isn't in the golang function.

Reply:
Right. You asked me to simplify this whole function and I identified that the last restart date was being repeated, so I abstracted a couple of blocks with this function. It's small enough that we don't lose a lot of connection with the original function, as we would if we try to abstract the much more complex two container blocks.